File: examples/boxlite/src/setup-image.ts
import { execSync } from "node:child_process";
import { existsSync, mkdirSync } from "node:fs";

export const DOCKER_IMAGE = "sandbox-agent-boxlite";
export const OCI_DIR = new URL("../oci-image", import.meta.url).pathname;

export function setupImage() {
	console.log(`Building image "${DOCKER_IMAGE}" (cached after first run)...`);
	execSync(`docker build -t ${DOCKER_IMAGE} ${new URL("..", import.meta.url).pathname}`, { stdio: "inherit" });

	if (!existsSync(`${OCI_DIR}/oci-layout`)) {
		console.log("Exporting to OCI layout...");
		mkdirSync(OCI_DIR, { recursive: true });
		execSync(`docker save ${DOCKER_IMAGE} | tar -xf - -C ${OCI_DIR}`, { stdio: "inherit" });
	}
}

File: examples/boxlite-python/setup_image.py
"""Build the sandbox-agent Docker image and export it to OCI layout."""

import os
import subprocess

DOCKER_IMAGE = "sandbox-agent-boxlite"
OCI_DIR = os.path.join(os.path.dirname(os.path.abspath(__file__)), "oci-image")


def setup_image() -> None:
    dockerfile_dir = os.path.dirname(os.path.abspath(__file__))

    print(f'Building image "{DOCKER_IMAGE}" (cached after first run)...')
    subprocess.run(
        ["docker", "build", "-t", DOCKER_IMAGE, dockerfile_dir],
        check=True,
    )

    if not os.path.exists(os.path.join(OCI_DIR, "oci-layout")):
        print("Exporting to OCI layout...")
        os.makedirs(OCI_DIR, exist_ok=True)
        subprocess.run(
            [
                "skopeo", "copy",
                f"docker-daemon:{DOCKER_IMAGE}:latest",
                f"oci:{OCI_DIR}:latest",
            ],
            check=True,
        )
